repositories/game.py

```diff
@@ -18,3 +18,7 @@ def get(self, id: int) -> Optional[Game]:
 
     def get_many(self, count: int) -> List[Game]:
         return self.db.query(Game).limit(count).all()
+
+    def get_available(self, count: int) -> List[Game]:
+        games = self.db.query(Game).filter(Game.started == False).all()
+        return [game for game in games if len(game.players) < game.max_players][:count]
```

Comment:
Saltate el paso de descargar todas las partidas para filtrar las que si queres ver.
(tenes que importar `sqlalchemy.func`)
```suggestion
    def get_available(self, count: int) -> List[Game]:
        q = (
            self.db.query(Game)
            .outerjoin(Game.players)
            .group_by(Game.id)
            .having(func.count(Player.id) < Game.max_players)
            .filter(Game.started == False)
        )
        if count is not None:
            q = q.limit(count)
        return q.all()
```
Es mas codigo pero es mucho mejor, en lugar de descargar todo filtras en la db

Reply:
> Saltate el paso de descargar todas las partidas para filtrar las que si queres ver. (tenes que importar `sqlalchemy.func`)
> 
> Es mas codigo pero es mucho mejor, en lugar de descargar todo filtras en la db

¡Muy buen uso de having! supongo que puedo hacer un `.order_by(func.count(Player.id))` para ordenarlos por cantidad de jugadores en lobby

Reply:
Sip, exactamente